packages/headless/src/controllers/recent-queries-list/headless-recent-queries-list.ts

```diff
@@ -178,7 +180,9 @@ export function buildRecentQueriesList(
       if (errorMessage) {
         throw new Error(errorMessage);
       }
+      dispatch(deselectAllFacets());
       dispatch(updateQuery({q: this.state.queries[index]}));
+      dispatch(updatePage(1));
       dispatch(executeSearch(logRecentQueryClick()));
```

Comment:
This will work well, however if we want to guarantee the two controllers remain in sync as the project evolves, it might be worthwhile to extract the common actions into a thunk action [similar to here](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/features/facets/facet-set/facet-set-controller-actions.ts#L24).

Not sure what to name the thunk though ... maybe `searchWithQuery`? Since this sequence of actions is needed when changing the query.

```
dispatch(searchWithQuery({q: '...', analytics: ...}))
```
Another approach could be to keep the `executeSearch` outside of the thunk to avoid needing to pass analytics in.

```
dispatch(prepareForSearchWithQuery(...some query...));
dispatch(executeSearch(analytics));
```